plugins/tokenVoting/pages/new.tsx

```diff
@@ -79,7 +68,12 @@ export default function Create() {
         const blob = new Blob([JSON.stringify(proposalMetadataJsonObject)], { type: 'application/json' });
 
         const ipfsPin = await uploadToIPFS(ipfsClient, blob);
-        setIpfsPin(ipfsPin!)
+        createProposalWrite({
```

Comment:
await? 
Get the hash and check the TX result?

Reply:
What do you mean? Not sure I follow the questions :)

The await in there is only for IPFS. The `createProposalWrite()` doesn't need await. Checkout line #34, the variable data is a reactive variable that has the tx hash. The result comes from there, not from this function.

Reply:
Not sure I follow
The code snippet starts a transaction but doesn't handle whether it succeeds or fails:

```ts
        createProposalWrite({
            abi: TokenVotingAbi,
            address: PLUGIN_ADDRESS,
            functionName: 'createProposal',
            args: [toHex(ipfsPin), actions, 0, 0, 0, 0, 0],
        })
```

A similar issue already exists if you reject and try to rerun the transaction again. Nothing will happen

Reply:
The same also applies to `delegateWrite` and `delegateAnnouncementWrite` in UserDelegateCard

Reply:
The handling of success doesn't happen there, but in the hook. That's why I pointed the line 34, because that's where I check that it succeeded and tx hash. 
That's how wagmi works.

Agree that I'm not checking if the users rejects the creation. We have a ticket in Linear for that tho. I'd suggest doing both together rather than one solution here and another in the other plugin.